server/routes/prisons/prisons.test.ts

```diff
@@ -46,3 +49,65 @@ describe('GET /prisons', () => {
       })
   })
 })
+
+describe('POST /prisons', () => {
+  it('should add prison to list of supported prisons and display message', () => {
+    return request(app)
+      .post('/prisons')
+      .send('prisonId=BWI')
+      .expect(302)
+      .expect('location', `/prisons`)
+      .expect(() => {
+        expect(flashProvider).toHaveBeenCalledWith('message', 'BWI')
+        expect(prisonService.createPrison).toHaveBeenCalledTimes(1)
+        expect(prisonService.createPrison).toHaveBeenCalledWith('user1', 'BWI')
+      })
+  })
+})
+
+describe('GET /prisons/HEI/edit', () => {
+  it('should display correct prison specific information and offer correct action', () => {
+    return request(app)
+      .get('/prisons/HEI/edit')
+      .expect('Content-Type', /html/)
+      .expect(res => {
+        const $ = cheerio.load(res.text)
+        expect($('h1').text().trim()).toBe(prisonName)
+
+        expect($('[data-test="prison-name-status"]').eq(0).text()).toContain(prisonName)
+        expect($('[data-test="prison-name-status"]').eq(0).text()).toContain('Active')
+
+        expect($('.govuk-button').text()).toContain('Deactivate')
+      })
+  })
+})
+
+describe('POST /prisons/HEI/edit', () => {
+  it('should change prison status and display message when submit (deactivate)', () => {
+    return request(app)
+      .post('/prisons/HEI/edit')
+      .send('action=deactivate')
+      .expect(302)
+      .expect('location', `/prisons/HEI/edit`)
+      .expect(() => {
+        expect(flashProvider).toHaveBeenCalledWith('message', 'deactivated')
+        expect(prisonService.deactivatePrison).toHaveBeenCalledTimes(1)
+        expect(prisonService.deactivatePrison).toHaveBeenCalledWith('user1', 'HEI')
+        expect(prisonService.activatePrison).toHaveBeenCalledTimes(0)
+      })
+  })
+
+  it('should change prison status and display message when submit (deactivate)', () => {
```

Comment:
typo ('deactivate'; should be 'activate'!) in description - but otherwise looks good 🙂!